### DISCUSSION TITLE
12.1.0

### DISCUSSION THREAD

**Discussion by PierfrancescoSoffritti** [PierfrancescoSoffritti/android-youtube-player#1038] [Announcements]
This release fixes a few bugs and improves support for playlists.

* #460 and #887 are fixed. Now `YouTubePlayer` has more methods to control the playback of playlists. Such as `nextVideo`, `previousVideo`, `playVideoAt`, `setLoop` and `setShuffle`.
* #996 is fixed, thanks to @maharshpatel247
* #929 is fixed, modest branding can be set to 0, thanks to @Berdorge
* #860 is fixed.

**Sponsors**

For the first time ever someone sponsored the project, by making a donation. Thank you!
* @RedaGhoussoub
* @ben-WIM

**New Contributors**
* @maharshpatel247 made their first contribution in #996
* @Berdorge made their first contribution in #1005
* @MananArya22 made their first contribution in #951

**Full Changelog**: https://github.com/PierfrancescoSoffritti/android-youtube-player/compare/12.0.0...12.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/PierfrancescoSoffritti/android-youtube-player/releases/tag/12.1.0'>12.1.0</a>.</em>

**Comment by billdizl** [comment]
12.0.0 and 12.1.0 version have a lot bug of this lib

**Comment by Anuwatrodcharoen** [comment]
> This release fixes a few bugs and improves support for playlists.
> 
> * #460 and #887 are fixed. Now `YouTubePlayer` has more methods to control the playback of playlists. Such as `nextVideo`, `previousVideo`, `playVideoAt`, `setLoop` and `setShuffle`.
> * #996 is fixed, thanks to @maharshpatel247
> * #929 is fixed, modest branding can be set to 0, thanks to @Berdorge
> * #860 is fixed.
> 
> **Sponsors**
> 
> For the first time ever someone sponsored the project, by making a donation. Thank you!
> * @RedaGhoussoub
> * @ben-WIM
> 
> **New Contributors**
> * @maharshpatel247 made their first contribution in #996
> * @Berdorge made their first contribution in #1005
> * @MananArya22 made their first contribution in #951
> 
> **Full Changelog**: https://github.com/PierfrancescoSoffritti/android-youtube-player/compare/12.0.0...12.1.0
> 
> <hr /><em>This discussion was created from the release <a href='https://github.com/PierfrancescoSoffritti/android-youtube-player/releases/tag/12.1.0'>12.1.0</a>.</em>
